toolchain/sem_ir/inst.h

```diff
@@ -124,8 +118,7 @@ struct InstLikeTypeInfo<
 //   data where the instruction's kind is not known.
 class Inst : public Printable<Inst> {
  public:
-  template <typename TypedInst,
-            typename Info = typename InstLikeTypeInfo<TypedInst>::Self>
+  template <InstLikeType TypedInst>
```

Comment:
The style guide suggests to avoid this syntax, and prefers something like this from my reading:

```suggestion
  template <typename TypedInst>
    requires InstLikeType<TypedInst>
```

Reply:
Done.